_blog/2018/soft-multitenancy.md

```diff
@@ -0,0 +1,298 @@
+tenants is marginal compared to sharing a kubernetes cluster across tenants.
+
+Considering aforementioned challenges, this blog will describe how to provide Option 3.
+Current Istio capabilities are well suited to providing this option as namespace based scoping
+is already widely supported in most Istio modules. Best practices for deploying multiple
+tenant applications per cluster require the use of a namespace. This blog will provide a
+high level description on what is required to deploy multiple Istio control planes
+(one per tenant) on a single Kubernetes cluster.
+
+## Deployment details
+#### Multiple Istio control planes
+Deploying multiple Istio control planes starts by replacing all `namespace` references
+in a manifest file with the desired namespace. Using istio.yaml as an example, if two tenant
+level istio control planes are required; the first can use the istio.yaml default name of
+*istio-system* and a second control plane can be created by generating a new yaml file with
+a different namespace.
+```bash
+cat istio.yaml | sed s/istio-system/istio-system1/g > istio-system1.yaml
+```
+Note that the execution of these two yaml files is the responsibility of the cluster
+administrator, not the tenant level adminstrator. Additional RBAC restrictions will also
+be configured and applied by the cluster administator limiting the tenant administrator to only
+the assigned namespace.
+
+If the Istio [addons](https://istio.io/docs/tasks/telemetry/) are required then the manifests must
+be updated to match the configured `namespace` in use by the tenant's Istio control plane.
+
+#### Split Common and Namespace Specific Resources
+The manifest files in the Istio repositories create both common resources that would
+be used by all Istio control planes as well as resources that are replicated per control
+plane. Although it is a simple matter to deploy multiple control plane by replacing the
+*istio-system* namespace references as described above, a better approach is to split the
+manifests into a common part that is deployed once for all tenants and a per tenant
+specific portion. All the CustomResourceDefinitions (CRDs), the roles and the role
+bindings should be separated out from the provided Istio manifests.  Additionally, the
+roles and role bindings in the provided Istio manifests are probably unsuitable for a
+multitenant environment and should be modified or augmented as described in the next
+section.
+
+#### Kubernetes RBAC for Istio control plane resources
+To restrict a tenant administrator to a single Istio namespace, the cluster
+administrator would create a manifest containing, at a minimum, a `Role` and `RoleBinding`
+similar to the one below. In this example, a tenant administrator named *sales-admin*
+is limited to the namespace *istio-system1*. A completed manifest would contain many
+more `apiGroups` under the `Role` providing resouce access to the tenant administator.
+```yaml
```

Comment:
Have a review of: http://www.yaml.org/spec/1.2/spec.html#id2760395 - this may render fine, but do these examples work as is?  I'm not sure what ```yaml does in the jekyll rendering (i.e. does it make the yaml compliant with the yaml 1.2 spec?)